internal/pkg/aws/resourcegroups/resourcegroups.go

```diff
@@ -0,0 +1,14 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package resourcegroups provides a client to make API requests to AWS Resource Groups
+
+package resourcegroups
+
+import (
+	"github.com/aws/aws-sdk-go/service/resourcegroups"
+)
+
+type ResourceGroupClient interface {
+	SearchResources(input *resourcegroups.SearchResourcesInput) (*resourcegroups.SearchResourcesOutput, error)
+}
```

Comment:
Should we provide a client here and hide the SDK interface? Otherwise consumers of this package needs to import the SDK's resourcegroups package as well and have to repeat a similar way of querying resource groups.

```go
type api interface {
  SearchResources(input *resourcegroups.SearchResourcesInput) (*resourcegroups.SearchResourcesOutput, error)
}

type ResourceGroups struct {
  client api
}

// internally sets the type to TAG_FILTERS_1_0 and generates the query with resource type and tags.
func (rg *ResourceGroups) SearchResourcesByTags(resourceType string, tags map[string]string) (arns []string, error)
```

Reply:
I did that originally but it that would have required a much bigger change in all the dependencies and seemed unnecessary for the current use case, since right now the client just gets instantiated with a "New" as a field in another interface (see: https://github.com/aws/amazon-ecs-cli-v2/blob/master/internal/pkg/aws/cloudwatch/cloudwatch.go#L54). For sake of agility I figured we can add all those wrappers when we needed to access ResourceGroups as a standalone client

Reply:
I don't think I follow the argument, the `rgClient` in the cw package uses the `SearchResources` method:
https://github.com/aws/amazon-ecs-cli-v2/blob/62e5515b4b6ce53173919b665404fc0fd1b76042/internal/pkg/aws/cloudwatch/cloudwatch.go#L67
If I understand correctly, the new `codepipeline` pkg will also have to use the same method and the same way of querying the resources. So it made sense to me to move the method to this package.

Reply:
The `SearchResourceByTags` method refers to some cloudwatch specific things (eg https://github.com/aws/amazon-ecs-cli-v2/blob/62e5515b4b6ce53173919b665404fc0fd1b76042/internal/pkg/aws/cloudwatch/cloudwatch.go#L62). I understand the argument about not wanting to import the sdk as a dependency, but like I said i was trying to change as few things as possible, esp in the cloudwatch package for now, so that I could focus on the pipeline work that needs this and extract the method/implementation of the resource group client separately. But if you prefer that work to be done now than I can do that.

Reply:
FIXED